android/clientlib-ktx/src/main/java/com/solana/mobilewalletadapter/clientlib/MobileWalletAdapter.kt

```diff
@@ -0,0 +1,102 @@
+package com.solana.mobilewalletadapter.clientlib
+
+import android.util.Log
+import com.solana.mobilewalletadapter.clientlib.protocol.JsonRpc20Client
+import com.solana.mobilewalletadapter.clientlib.protocol.MobileWalletAdapterClient
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationIntentCreator
+import com.solana.mobilewalletadapter.clientlib.scenario.LocalAssociationScenario
+import com.solana.mobilewalletadapter.clientlib.scenario.Scenario
+import com.solana.mobilewalletadapter.common.ProtocolContract
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import java.io.IOException
+import java.util.concurrent.CancellationException
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeoutException
+
+class MobileWalletAdapter(
+    private val timeout: Int = Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
+    private val ioDispatcher: CoroutineDispatcher = Dispatchers.IO
+) {
+
+    private val adapterOperations = LocalAdapterOperations(ioDispatcher)
+
+    suspend fun <T> transact(sender: ActivityResultSender, block: suspend AdapterOperations.() -> T) {
```

Comment:
What would this API look like for different association scenarios other than `LocalAssociationScenario`?

```
// Basic, local.
transact(sender) { it.authorize(...) }

// Remote?
transact.withRemoteWallet(sender) { it.authorize(...) }
```

Or would it be more like…

```
transact(scenario, sender) { it.authorize(...) }
```